block/manager.go

```diff
@@ -277,7 +281,7 @@ func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
 			return fmt.Errorf("failed to save block responses: %w", err)
 		}
 
-		if m.daHeight > newState.DAHeight {
```

Comment:
Was that a bug you accidentally caught? 😁

Reply:
Yes!